VISION.md

```diff
@@ -0,0 +1,98 @@
+
+Provide rules to limit what language features can be used, including:
+
+-   Enforcing a maximum specificity by limiting the specificity itself or the occurrence of different selector types e.g. class, id, attribute etc.
+-   Enforcing best practices _at the configuration level_ e.g. disallowing the `all` keyword for transitions as it is not performant.
+-   Enforcing the use of a subset of features to improve consistency across a code base e.g. limiting what units are allowed (`px` or `rem` etc.)
+-   Enforcing specific patterns for selectors and names (e.g. those of custom properties).
+
+### Stylistic issues
+
+Provide rules to enforce a diverse range of stylistic conventions, including:
+
+-   Whitespace
+-   Case
+-   Quotes
+
+There are two approaches to enforcing stylistic conventions:
+
+-   A machine algorithmically pretty prints the code (usually based on a maximum line length).
+-   A human initially formats the code, and a machine fixes-up/warns-about any mistakes.
+
+The former is handled by pretty printers, like [prettier](https://github.com/prettier/prettier), whereas the latter is catered for by the built-in stylistic rules.
+
+Additionally, the built-in stylistic rules and plugins are configurable to support this diverse range of stylistic conventions. This is in contrast to pretty printers, which tend to be opinionated. The ordering of properties within declaration blocks is an example of a divisive topic, where there is no one or two dominant conventions. The [`stylelint-order`](https://www.npmjs.com/package/stylelint-order) plugin adheres to stylelint's philosophies, and can be used to lint and fix a diverse range of ordering conventions.
+
+Another example is the use of single line rules for sets of _related_ rules e.g.
+
+```css
+/* Single line related classes */
+.class-1 { top: 0; bottom: 0; }
+.class-2 { top: 5px; right: 0; }
+.class-3 { top: 8px; left: 0; }
+```
+
+The built-in stylistic rules can be configured to allow both multi-line and single-line rules. The choice of when to use each belongs to the user, who can determine which rules are related.
+
+## Extensible
+
+Provide multiple points of extensions, including:
+
+-   [Plugins](docs/developer-guide/plugins.md) - build community rules to support methodologies, toolsets, non-standard CSS features, or very specific use cases.
+-   [Processors](docs/user-guide/processors.md) - lint the CSS within non-stylesheet files.
+-   [Extendable configs](docs/user-guide/configuration.md#extends) - extend and share configurations.
+-   [Formatters](docs/developer-guide/formatters.md) - format stylelint result objects.
+-   [Custom syntax](docs/user-guide/node-api.md#customsyntax) - use any PostCSS-compatible syntax module.
+
+## Robust
+
+Provide a robust tool with a [comprehensive test suite](docs/developer-guide/rules.md#write-tests), including:
+
+-   High coverage, currently over 95%.
+-   A wide range of fixtures for rules, currently over 40000.
```

Comment:
Where this number came from? Jest is reporting only 12 805 tests on `v8` branch.